p2p/src/peer_manager.rs

```diff
@@ -82,36 +89,63 @@ where
     fn dialable_peers(&self) -> Vec<SocketAddr> {
         self.candidate_peers
             .iter()
-            .filter(|peer| !self.connected_peers.contains_key(peer))
+            .filter_map(|(peer, address)| {
+                (!self.connected_peers.contains_key(peer)).then(|| address.clone())
+            })
             .take(self.max_peers as usize - self.connected_peers.len())
-            .cloned()
             .collect()
     }
 
-    async fn new_peer(&mut self, peer: SocketAddr, mut stream: TcpStream) {
+    async fn new_peer(&mut self, address: SocketAddr, mut stream: TcpStream) {
```

Comment:
Note: we need to know peer's index in the committee somehow for signature aggregation in the future.

Reply:
Indeed! Each node should have the same list of (ValidatorIndex, PublicKey). This will be provided via config and I'll work on that part in the following PR!